lib/jxl/dec_patch_dictionary.cc

```diff
@@ -177,44 +177,138 @@ int PatchDictionary::GetReferences() const {
   return result;
 }
 
-void PatchDictionary::ComputePatchCache() {
-  patch_starts_.clear();
-  sorted_patches_.clear();
-  if (positions_.empty()) return;
-  std::vector<std::pair<size_t, size_t>> sorted_patches_y;
-  for (size_t i = 0; i < positions_.size(); i++) {
-    const PatchPosition& pos = positions_[i];
-    const PatchReferencePosition& ref_pos = ref_positions_[pos.ref_pos_idx];
-    for (size_t y = pos.y; y < pos.y + ref_pos.ysize; y++) {
-      sorted_patches_y.emplace_back(y, i);
-    }
+namespace {
+struct PatchInterval {
```

Comment:
Does it make sense to extract the interval tree in a class?

Reply:
Maybe if we use it for something else too.